### PR TITLE
add ide files to gitignore

### PR DESCRIPTION
**Pull request by Disservin**
I think it would be great for any current/new dev's if we add some more IDE related paths to the gitignore.
Feel free to propose additions

No functional change

### CODE REVIEW

**Comment by UniQP** [comment]
I think IDE files should be ignored by the users themselves. There are many different IDEs and we would need to maintain the list of ignored IDE files in all repositories.

The proper way of ignoring IDE files (according to my knowledge) is to use git's `core.excludesfile` config to specify a user-specific ignore file that applies to all repositories. I would rather mention the existence of this git config somewhere than actually adding IDE-specific files to gitignore.

**Comment by Disservin** [comment]
Mhh, good point, regarding the `core.excludesfile`. 
Although when you have a look at a few of the most popular repositories on Github, they kinda all do this. 
I agree that I wouldnt want to update the gitignore every few months because of a new ide, but some of the most popular ones might be helpful?

I had a look at: 
https://github.com/vitejs/vite/blob/main/.gitignore
https://github.com/laravel/laravel/blob/10.x/.gitignore
https://github.com/vuejs/vue/blob/main/.gitignore
https://github.com/tensorflow/tensorflow/blob/master/.gitignore
https://github.com/electron/electron/blob/main/.gitignore
https://github.com/tauri-apps/tauri/blob/dev/.gitignore
https://github.com/facebook/react/blob/main/.gitignore

all of these pretty much at least add vscode/idea and .DS_STORE which is mac related.

**Comment by ddobbelaere** [comment]
Just adding to the suggestion of @UniQP, it's also possible to have other `.gitignore` files in the tree. E.g. a user might put a `.gitignore` inside the `.vscode` directory that ignores all files inside it. This is how e.g. mypy does it with their `.mypy_cache` directory and how some VSCode extensions that create build directories do it.

That being said, I'm not against this PR if it's only for the most popular IDEs.

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,14 @@ src/-lstdc++.res
 # Neural network for the NNUE evaluation
 **/*.nnue
 
+# IDE's
```
Comment (ddobbelaere) [inline .gitignore:13]: IDEs is the correct spelling.

**Comment by Disservin** [comment]
If the common consens is that the user should specify this then I have no problem with that and we can then close this.

**Comment by vondele** [comment]
I think developer/IDE specific files are probably better handled differently.

**Pull request closed without merging**